src/cryptonote_basic/account.cpp

```diff
@@ -54,7 +55,12 @@ DISABLE_VS_WARNINGS(4244 4345)
 
   namespace cryptonote
 {
-
+  static void generate_ed25519_keys(account_keys &keys)
+  {
+    crypto::secret_key const &spend_key          = keys.m_spend_secret_key;
+    crypto::ec_scalar const &spend_key_unwrapped = unwrap(unwrap(keys.m_spend_secret_key));
+    crypto_sign_ed25519_seed_keypair(keys.m_spend_ed25519_public_key.data, keys.m_spend_ed25519_secret_key.data, reinterpret_cast<unsigned char const *>(spend_key_unwrapped.data));
```

Comment:
You don't need to do any type unwrapping here; `secret_key` is required to be directly copyable:
```C++
crypto_sign_ed25519_seed_keypair(
    keys.m_spend_ed25519_public_key.data,
    keys.m_spend_ed25519_secret_key.data,
    reinterpret_cast<unsigned char const *>(&keys.m_spend_secret_key));
```